segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -592,7 +599,14 @@ public void mergeSegmentsBatch(WireCommands.MergeSegmentsBatch mergeSegments) {
                         throw new CompletionException(e);
                     } else {
                         recordStatForTransaction(r, mergeSegments.getTargetSegmentId());
-                        indexAppendProcessor.processAppend(mergeSegments.getTargetSegmentId());
+                        try {
+                            SegmentProperties mergeSegmentProp = segmentStore.getStreamSegmentInfo(mergeSegments.getTargetSegmentId(), TIMEOUT).join();
+                            long eventCount = mergeSegmentProp.getAttributes().get(EVENT_COUNT) != null ? mergeSegmentProp.getAttributes().get(EVENT_COUNT) : 0L;
```

Comment:
This is wrong. It is passing the event count from the segment being merged, not the segment being merged into. This should instead be handled internal to the indexAppendProcessor.